### PR TITLE
Submenu

### PR DESCRIPTION
**Pull request by onsah**
Adds the submenu feature described in #10 
Some highlights:
* Recursively nested submenus. There is no limit on the number of levels of submenus
* Customizable indicator for submenus
* Automatic creation and removal of submenus when neccessary (Such as inserting the first item or removing the last item in the submenu)

For the details refer to [the documentation](README.md) (mostly changes are in the api section)

Note: This PR should be merged after #47 because it is built on it

### CODE REVIEW

**Comment by onsah** [comment]
@hasanbalci This PR is now ready to be merged.

File: README.md [hunk 1]
```diff
@@ -86,6 +89,7 @@ var options = {
         show: false, // Whether the item will be shown or not
         hasTrailingDivider: true, // Whether the item will have a trailing divider
         coreAsWell: false // Whether core instance have this item on cxttap
+        submenu: [] // Shows the listed menuItems as a submenu for this item. An item can't have both submenu and onClickFunction property at once
```
[the remark this answers is not in the document]
Comment (onsah) [reply README.md]: Yes it is incorrect. It is changed to an item must have at least either one or both.

**Comment by onsah** [comment]
> A few comments for now:
> 
>     * For example, I have
> 
> 
> ```
> blue -> light blue
> green
> ```
> 
> and if I move light blue to parent menu by using `moveToSubmenu` function then submenu indicator still exists even though there is no submenu anymore.
> 
>     * In the demo, if I click on 'change color' item, I get error in console.

Thanks for noticing this issue. It is caused because I changed `moveToSubmenu` just detaching from the parent instead of removing it, because when removing a menu item is also destroyed. But this is a problem for moving items that contain a submenu. I am trying to find a good solution.

For the second issue, what kind of error do you get? Can you provide the error because I don't get any error in the console when I click 'change-color'.

**Comment by onsah** [comment]
I have fixed the problems and it should be ready to merge now.

File: README.md [hunk 2]
```diff
@@ -120,52 +124,76 @@ var options = {
...
-instance.setTrailingDivider(itemID, status); // Sets whether the menuItem with given ID will have a following divider.
+#### `instance.insertBeforeMenuItem(item, existingItemID)`
+* Inserts given item before the existingitem
 
-instance.insertBeforeMenuItem(item, existingItemID); // Inserts given item before the existingitem.
+#### `instance.moveToSubmenu(itemID, options = null)`
+* Moves the item with given ID to the submenu of the parent with the given ID or to root with the specified options
+* If `options` is a `string`, then it is the id of the parent
+* If `options` is a `{ selector?: string, coreAsWell?: boolean }`, then old properties are overwritten by them and the menu item is moved to the root. If it doesn't have either properties item is **not moved**.
+* If `options` in null or not provided, then it is just moved to the root
```
Comment (hasanbalci) [inline README.md:162]: in -> is

File: README.md [hunk 3]
```diff
@@ -120,52 +124,76 @@ var options = {
...
+#### `instance.moveToSubmenu(itemID, options = null)`
+* Moves the item with given ID to the submenu of the parent with the given ID or to root with the specified options
+* If `options` is a `string`, then it is the id of the parent
+* If `options` is a `{ selector?: string, coreAsWell?: boolean }`, then old properties are overwritten by them and the menu item is moved to the root. If it doesn't have either properties item is **not moved**.
+* If `options` in null or not provided, then it is just moved to the root  
 
-instance.moveBeforeOtherMenuItem(itemID, existingItemID); // Moves the item with given ID before the existingitem.
+#### `instance.moveBeforeOtherMenuItem(itemID, existingItemID)`
+* Moves the item with given ID before the existingitem.
+* If the are not in the same submenu, item with the given ID will be moved to the submenu that contains the existingItem
```
[the remark this answers is not in the document]
Comment (onsah) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (onsah) [reply README.md]: Fixed

**Pull request merged**